src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/DefaultProcessConfigurationTests.cs

```diff
@@ -0,0 +1,178 @@
+            };
+
+            var filterDescriptorCommandContains = new ProcessFilterDescriptor
+            {
+                Key = ProcessFilterKey.CommandLine,
+                MatchType = ProcessFilterType.Contains,
+                Value = "arg1"
+            };
+
+            var filter = CreateFilterEntry(filterDescriptorPid);
+            ValidateProcessFilter(DiagProcessFilterCriteria.ProcessId, filterDescriptorPid.Value, filter);
+
+            //Contains still becomes Exact for ProcessId
+            filter = CreateFilterEntry(filterDescriptorPidContains);
+            ValidateProcessFilter(DiagProcessFilterCriteria.ProcessId, filterDescriptorPidContains.Value, filter);
+
+            filter = CreateFilterEntry(filterDescriptorName);
+            ValidateProcessFilter(DiagProcessFilterCriteria.ProcessName, filterDescriptorName.Value, filter);
+
+            filter = CreateFilterEntry(filterDescriptorNameContains);
+            ValidateProcessFilter(DiagProcessFilterCriteria.ProcessName, filterDescriptorNameContains.Value, DiagProcessFilterMatchType.Contains, filter);
+
+            filter = CreateFilterEntry(filterDescriptorCommand);
+            ValidateProcessFilter(DiagProcessFilterCriteria.CommandLine, filterDescriptorCommand.Value, filter);
+
+            filter = CreateFilterEntry(filterDescriptorCommandContains);
+            ValidateProcessFilter(DiagProcessFilterCriteria.CommandLine, filterDescriptorCommandContains.Value, DiagProcessFilterMatchType.Contains, filter);
+
+            //This filter doesn't make any sense but we are just testing that we can combine multiple filters
+            var options = CreateOptions(filterDescriptorPid, filterDescriptorName, filterDescriptorNameContains, filterDescriptorCommand, filterDescriptorCommandContains);
+
+            ValidateProcessFilter(DiagProcessFilterCriteria.ProcessId, filterDescriptorPid.Value, options.Filters[0]);
+            ValidateProcessFilter(DiagProcessFilterCriteria.ProcessName, filterDescriptorName.Value, options.Filters[1]);
+            ValidateProcessFilter(DiagProcessFilterCriteria.ProcessName, filterDescriptorNameContains.Value, DiagProcessFilterMatchType.Contains, options.Filters[2]);
+            ValidateProcessFilter(DiagProcessFilterCriteria.CommandLine, filterDescriptorCommand.Value, options.Filters[3]);
+            ValidateProcessFilter(DiagProcessFilterCriteria.CommandLine, filterDescriptorCommandContains.Value, DiagProcessFilterMatchType.Contains, options.Filters[4]);
+        }
+
+        [Fact]
+        public void NewCriteriaTest()
+        {
+            //When new enumerations (such as Entrypoint) are added to DiagProcessFilterCriteria,
+            //this test will fail. This means new unit tests should be written for that criteria and this test can be updated.
+
+            //Sorted by unsigned integer value
+            var expectedValues = Enum.GetValues(typeof(DiagProcessFilterCriteria));
+
+            var actualValues = new[]
```

Comment:
nit: These are the expected values and the ones from Enum.GetValues are the actual values.

Reply:
I could see that, but I guess I don't think of it that way in this case...the expected case is that we have test coverage for everything in the enum. The actual values represent what we've actually written tests for. I'll leave this alone for now.

Reply:
What we expect is the enum to have exactly those four values. What it actually has is found in the result of Enum.GetValues. Hence, when we add another enum value, we still expected four items in the test and actually have five.